libraries/access.rb

```diff
@@ -70,7 +70,7 @@ class PgHbaFile
 
           attr_reader :entries
 
-          SPLIT_REGEX = %r{^(?<type>\w+)\s+(?<database>\w+)\s+(?<user>\w+)\s+(?<address>[\d.:\/]+)?(?:\s*)(?<auth_method>[\w-]+)(?:\s*)(?<auth_options>[\w=-]+)?(?:\s*)(?<comment>#\s*.*)?$}.freeze
+          SPLIT_REGEX = %r{^(((?<type>local)\s+(?<database>\w+)\s+(?<user>\w+))|((?!local)(?<type>\w+)\s+(?<database>\w+)\s+(?<user>\w+)\s+(?<address>[\w\-.:\/]+)))\s+(?<auth_method>[\w-]+)(?:\s*)(?<auth_options>[\w=-]+)?(?:\s*)(?<comment>#\s*.*)?$}.freeze
```

Comment:
If we're just looking to match a hostname or fqdn in the address field is there any reason we can't just add `\w` to the address capture group character class?

ie:
```ruby
^(?<type>\w+)\s+(?<database>\w+)\s+(?<user>\w+)\s+(?<address>[\w\d.:\/]+)?(?:\s*)(?<auth_method>[\w-]+)(?:\s*)(?<auth_options>[\w=-]+)?(?:\s*)(?<comment>#\s*.*)?$
```

I'm not a big fan of making the regex any more complicated than it absolutely needs to be and as long as it (reasonably, it is a regex afterall) reliably breaks up the entry I'm happy with it. If we needed any validation fields etc. they'd be best implemented in the class representing an entry instead.

Reply:
This my first thought, too. But this makes the expression ambiguous, as it doesn't skip the `address` field for `local`-rules. See:

```
irb(main):002:0> %r'^(?<type>\w+)\s+(?<database>\w+)\s+(?<user>\w+)\s+(?<address>[\w\d.:\/]+)?(?:\s*)(?<auth_method>[\w-]+)(?:\s*)(?<auth_op
tions>[\w=-]+)?(?:\s*)(?<comment>#\s*.*)?$'.match('local   foo     postgres    peeer').named_captures.compact.transform_keys(&:to_sym)
=> {:type=>"local", :database=>"foo", :user=>"postgres", :address=>"peee", :auth_method=>"r"}
```

Reply:
Missed that, apologies. Bloody regex strikes again. I'm still not **loving** it but might be the best we can do, I tried my best to avoid using a regex in the first place but dealing with comments made it a requirement.